core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStore.java

```diff
@@ -271,11 +271,18 @@ private void createSchemaTopic() throws StoreInitializationException {
     }
   }
 
-  static List<String> brokersToEndpoints(List<Broker> brokers) {
-    List<String> endpoints = new LinkedList<String>();
+  static List<String> brokersToEndpoints(List<Broker> brokers, String securityProtocol) {
+    List<String> endpoints = new LinkedList<>();
     for (Broker broker : brokers) {
       for (EndPoint ep : JavaConversions.asJavaCollection(broker.endPoints())) {
-        endpoints.add(ep.connectionString());
+        String hostport = ep.host() == null ? ":" + ep.port() : Utils.formatAddress(ep.host(), ep.port());
+        String endpoint = ep.securityProtocol() + "://" + hostport;
+        if (!ep.securityProtocol().equals(securityProtocol)) {
```

Comment:
It does seem weird to configure `KAFKASTORE_BOOTSTRAP_SERVERS` with their protocols only to ignore them.